pkg/machines/components/create-vm-dialog/pxe-helpers.js

```diff
@@ -0,0 +1,187 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import cockpit from 'cockpit';
+
+import * as Select from 'cockpit-components-select.jsx';
+
+const _ = cockpit.gettext;
+
+/**
+ * Returns a list of all virbrX Virtual Networks.
+ * @param {array} virtualNetworks - An array of object containing Virtual Networks.
+ */
+function getLibvirtNetworkBridges(virtualNetworks) {
+    return virtualNetworks
+            .filter(network => network.bridge)
+            .map(network => network.bridge.name);
+}
+
+/**
+ * Filters an array of node devices returning only devices of specific connection and capability.
+ * @param {array} nodeDevices - An array of object containing NodeDevices.
+ * @param {string} connectionName - The libvirt connection name.
+ * @param {string} type - The capability type, ex 'net'.
+ */
+function getNodeDevicesOfType(nodeDevices, connectionName, type) {
+    return nodeDevices.filter(nodeDevice => {
+        const capabilityType = nodeDevice.capability.type;
```

Comment:
This is unnecessary boilerplate, the const is being used exactly once. (But matter of preference, I figure).